Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -107,16 +108,9 @@ protected AbilityImpl(final AbilityImpl ability) {
         this.manaCosts = ability.manaCosts.copy();
         this.manaCostsToPay = ability.manaCostsToPay.copy();
         this.costs = ability.costs.copy();
-        for (Watcher watcher : ability.getWatchers()) {
-            watchers.add(watcher.copy());
-        }
+        this.watchers = CardUtil.deepCopyList((List)ability.getWatchers());
```

Comment:
curious why the cast to (List) is needed?

Reply:
I was using type-erasure to Object, making it necessary. However, I can instead use Generics in the deep copy code, so the only unchecked conversions like that happen inside the `deepCopyObject` function. This makes it look cleaner overall so I'm switching the code to that.